### PR TITLE
New markup structure and styles to support single and dual input rows with one or two control buttons

### PR DESCRIPTION
**Pull request by sg00dwin**
Fixes https://github.com/openshift/origin-web-console/issues/2858

- Consistent label presentation above form rows for secrets, webhooks, env-from, key-value

localhost_9000_dev-console_project_myproject_edit_builds_rails-pgsql-persistent 1024x768 view

screen shot 2018-03-08 at 10 39 12 am
screen shot 2018-03-08 at 10 38 45 am
screen shot 2018-03-08 at 10 38 07 am
screen shot 2018-03-08 at 10 36 57 am

### CODE REVIEW

**Comment by rhamilto** [comment]
@sg00dwin, were you able to cross-browser test?  I'm assuming not as I still don't have access to BrowserStack.  We should probably wait to review/merge until we can test?

**Comment by sg00dwin** [comment]
@rhamilto I was only able to test on Chrome, Safari, Firefox

File: app/styles/_forms.less [hunk 1]
```diff
@@ -1,3 +1,86 @@
...
+}
+.as-sortable-placeholder {
+  border-top: @as-sortable-placeholder-border-top-width @as-sortable-placeholder-border-top-style @as-sortable-placeholder-border-top-color;
+  clear: left;
+  width: @as-sortable-placeholder-width;
+}
+
+.form-row-has-controls {
+  position: relative;
+  width: calc(~"100% - 15px");
```
Comment (rhamilto) [inline app/styles/_forms.less]: 15px should be var to be consistent?

File: app/styles/_forms.less [hunk 2]
```diff
@@ -1,3 +1,86 @@
...
+      border: 0;
+      height: @form-row-control-button-height;
+      line-height: 1;
+      // match pf .close rule https://github.com/patternfly/patternfly/blob/master/src/less/close.less#L11
+      opacity: @as-sortable-dragging-opacity;
+      &:focus,
+      &:hover {
+        opacity: @as-sortable-item-icon-hover-opacity;
+      }
+      width: @form-row-control-button-width;
```
Comment (rhamilto) [inline app/styles/_forms.less]: Nit:  move these to line 62 so the new rule currently there doesn't interrupt this block of declarations.

File: app/styles/_variables.less [hunk 3]
```diff
@@ -22,26 +22,28 @@
...
 @btn-flat-bg: #f7f7f7;
 @btn-flat-border-color: #e7e7e7;
+@btn-close-icon-font-size: @font-size-large;
 @component-label: 10px;
 @console-panel-color: #fff;
 @console-text-color-default: @gray-dark;
 @easeInCubic: cubic-bezier(0.55, 0.055, 0.675, 0.19);
 @easeOutCubic: cubic-bezier(0.215, 0.61, 0.355, 1);
 @font-size-base: 12px;
+@form-row-control-button-height: @input-height-base;
```
Comment (rhamilto) [inline app/styles/_variables.less]: Nit:  alpha

File: app/styles/_variables.less [hunk 4]
```diff
@@ -22,26 +22,28 @@
...
+@form-row-control-button-gutter: 8px;
+@form-row-control-button-width: @input-height-base;
```
Comment (rhamilto) [inline app/styles/_variables.less]: Nit:  alpha

**Comment by spadgett** [comment]
@sg00dwin Looks like a dist mismatch

**Comment by sg00dwin** [comment]
Updated dist

**Pull request merged**